### PR TITLE
[@types/three] Adding MapControls from THREE r94

### PR DESCRIPTION
**Pull request by danyim**
This is a re-submission of #30104

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mrdoob/three.js/blob/dev/examples/js/controls/MapControls.js
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/three/three-mapcontrols.d.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    pan(deltaX: number, deltaY: number): void;
+
+    dollyIn(dollyScale: number): void;
+
+    dollyOut(dollyScale: number): void;
+
+    saveState(): void;
+
+    update(): false;
```
Comment (JulianSSS) [inline types/three/three-mapcontrols.d.ts]:
```suggestion
    update(): boolean;
```
MapControls.js returns true in line 216

**Review by JuStTheDev — CHANGES_REQUESTED** [review]
See line-comments

File: types/three/three-mapcontrols.d.ts (hunk 1 above)
Comment (danyim) [reply types/three/three-mapcontrols.d.ts]: Got it!

**Comment by danyim** [comment]
@JulianSSS Done!

**Review by JuStTheDev — APPROVED** [review]

**Comment by bhouston** [comment]
BTW we are trying to get *.d.ts files into Three.JS proper based on automated processing of the typings in types/three.  It would be good to have ownership of the *.d.ts files in Three.JS proper.  See here for more details: https://github.com/mrdoob/three.js/pull/15597

**Pull request merged**